polyply/src/check_residue_equivalence.py

```diff
@@ -42,3 +46,24 @@ def check_residue_equivalence(topology):
                 visited_residues[resname] = graph
                 molnames[resname] = mol_name
                 resids[resname] = molecule.nodes[node]["resid"]
+
+def group_residues_by_isomorphism(meta_molecule, template_graphs={}):
+    """
+    Collect all unique residue graphs. If the same resname matches
+    multiple graphs the resname is appended by a number. If required
+    template_graphs can be given that are used for matching rather
+    than the first founds residue.
+    """
+    unique_graphs = {}
+    for graph in template_graphs.values():
+        graph_hash = nx.algorithms.graph_hashing.weisfeiler_lehman_graph_hash(graph, node_attr='atomname')
+        unique_graphs[graph_hash] = graph
+
+    for node in meta_molecule.nodes:
+        graph = meta_molecule.nodes[node]["graph"]
+        graph_hash = nx.algorithms.graph_hashing.weisfeiler_lehman_graph_hash(graph, node_attr='atomname')
+        if graph_hash not in unique_graphs:
+            unique_graphs[graph_hash] = graph
+        meta_molecule.nodes[node]["template"] = graph_hash
```

Comment:
Or should the template be the graph itself?

Reply:
debatable; let's keep this for now otherwise it would mean changing too many downstream PRs